# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp
# Commit: f466599094b7
# Full Hash: f466599094b7e16948794a49b0571e03eb2b4df9
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-06-15 13:49:48
# Regressor Bug: 1962133
# File Overlap Count: 14
# Description:
#   Revert "Bug 1962133 - QM: Add accessed flag to FullOriginMetadata and persist it in the origin metadata file; r=dom-storage-reviewers,hsingh" as requested by dmeehan for causing top crash in Bug 1972221.
#   
#   This reverts commit 433768e64dd71c996694e08005854756b7c45fb3.
# ==============================================================================

diff -r 2408bee8bcc4 -r f466599094b7 dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp
--- a/dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp	Sat Jun 14 05:48:43 2025 +0000
+++ b/dom/quota/test/gtest/QuotaManagerDependencyFixture.cpp	Sun Jun 15 16:46:59 2025 +0300
@@ -6,7 +6,6 @@
 
 #include "QuotaManagerDependencyFixture.h"
 
-#include "DirectoryMetadata.h"
 #include "mozIStorageService.h"
 #include "mozStorageCID.h"
 #include "mozilla/BasePrincipal.h"
@@ -430,40 +429,6 @@
 }
 
 // static
-Maybe<OriginStateMetadata>
-QuotaManagerDependencyFixture::LoadDirectoryMetadataHeader(
-    const OriginMetadata& aOriginMetadata) {
-  auto result =
-      PerformOnIOThread([aOriginMetadata]() -> Maybe<OriginStateMetadata> {
-        QuotaManager* quotaManager = QuotaManager::Get();
-        MOZ_RELEASE_ASSERT(quotaManager);
-
-        auto directoryRes = quotaManager->GetOriginDirectory(aOriginMetadata);
-        MOZ_RELEASE_ASSERT(directoryRes.isOk());
-
-        auto directory = directoryRes.unwrap();
-
-        bool exists;
-        nsresult rv = directory->Exists(&exists);
-        MOZ_RELEASE_ASSERT(NS_SUCCEEDED(rv));
-
-        if (!exists) {
-          return Nothing();
-        }
-
-        auto originStateMetadataRes =
-            quota::LoadDirectoryMetadataHeader(*directory);
-        MOZ_RELEASE_ASSERT(originStateMetadataRes.isOk());
-
-        auto originStateMetadata = originStateMetadataRes.unwrap();
-
-        return Some(originStateMetadata);
-      });
-
-  return result;
-}
-
-// static
 uint64_t QuotaManagerDependencyFixture::TotalDirectoryIterations() {
   const auto result = PerformOnIOThread([]() -> uint64_t {
     QuotaManager* quotaManager = QuotaManager::Get();